modules/extension/wmts/src/test/java/org/geotools/ows/wmts/model/WMTSCapabilitiesTest.java

```diff
@@ -300,6 +300,20 @@ public void testParseWithIncompleteAddressMetadata() throws Exception {
         Assert.assertNotNull(address.getElectronicMailAddresses());
     }
 
+    /**
+     * The section OperationsMetadata of GetCapabilities is optional. WMTSCapabilities used as
+     * argument for WebMapTileServer shouldn't make a NullPointerException.
+     */
+    @Test
+    public void testMissingOperationsMetadata() throws Exception {
+        WMTSCapabilities capabilities = createCapabilities("linz_basemaps_capabilities.xml");
+        WebMapTileServer tileServer =
+                new WebMapTileServer(
+                        new URL("https://basemaps.linz.govt.nz/v1/tiles/WMTSCapabilities.xml"),
+                        capabilities);
+        Assert.assertNotNull(tileServer);
```

Comment:
Is this enough to prove the fix?works, original bug report indicated the error is thrown on requesting actual tiles not creation

Reply:
It's the constructor of WMTSCapabilities that fails, sending in serverURL=null. This test failed before I did my changes, but because of the deprecation I changed the constructor.
Not sure if that's the way to do it.